### PR TITLE
Document Half Open Range Patterns + Improve Range Patterns in General

### PR DESCRIPTION
**Pull request by Havvy**
This fulfills the reference PR requirement for https://github.com/rust-lang/rust/pull/102275

When looking at this PR, you will have to look at the commits individually. The first one swaps the style of the page to one sentence per line.

But also, before this can be merged, the question I raised in https://github.com/rust-lang/rust/issues/67264#issuecomment-1241356008 needs to be answered, although it does not substantially affect the rest of the prose w.r.t. a review.

### CODE REVIEW

File: src/patterns.md [hunk 1]
```diff
@@ -414,6 +389,7 @@ match tuple {
 >
 > _HalfOpenRangePattern_ :\
 > &nbsp;&nbsp; | _RangePatternBound_ `..`
+> &nbsp;&nbsp; | `..=` _RangePatternBound_
```
Comment (ehuss) [inline src/patterns.md:392]: Generally the grammar hasn't been using the leading-vert style (where the first alternation starts with `|`). I don't know why the original _HalfOpenRangePattern_ had it that way, it looks like an oversight. To stay consistent, could the first line change the `|` to `&nbsp;&nbsp;`?

File: src/patterns.md [hunk 2]
```diff
@@ -425,29 +401,51 @@ match tuple {
...
-not 0, and not negative numbers for signed integers. The bounds can be literals or paths that point
-to constant values.
+Range patterns are *half-open* if they have only an upper or lower bound.
+They have the same type as their upper or lower bound.
 
-A half-open range pattern in the style `a..` cannot be used to match within the context of a slice.
+A half open range with only a lower bound is written as its lower bound followed by `..`.
+These range patterns will match on any value greater than or equal to the lower bound.
+For example, `1..` will match 1, 9, or 9001, or 9007199254740991 (if it is of an appropriate size), but not 0, and not negative numbers for signed integers.
+For an integer the pattern `1..` will match 9, or 9001, or 9007199254740991 (if it is of an appropriate size), but not 0, and not negative numbers for signed integers.
```
Comment (ehuss) [inline src/patterns.md:428]: These two sentences look like duplicates of one another?

**Review by ehuss — COMMENTED** [review]
Thanks, looks great!

**Comment by ehuss** [comment]
If you want to post a PR with just the first commit, I can merge it now so it doesn't have to wait for anything.

**Comment by ehuss** [comment]
Looks good! I think the only open issue is the question about half-open ranges in slices.

**Comment by Havvy** [comment]
This PR left the original message about slice patterns in, so no change needs to be made due to that decision.

**Comment by Havvy** [comment]
cc @ehuss I think it is time to merge this.

**Review by ehuss — APPROVED** [review]
Oh, sorry, I missed that the PR merged. Thanks!

**Comment by Havvy** [comment]
Don't worry. I did too. Only saw it did going through the previous TWiR.

**Pull request merged**